### ISSUE TITLE
allow intel rings to be seen by everyone of a team

### ISSUE THREAD

**Issue by shalkya** [FAForever/fa#2654]
http://forums.faforever.com/viewtopic.php?f=41&t=13350

would enhance teamplay possibilities.

at very least making it for SMD, would be already nice. Also maybe give the option, so people that can't run it for the whole team, can choose to see only his intel rings.

**Comment by Garanas** [comment]
This requires an engine patch and was previously considered to be almost impossible to do. @KionX can you confirm that?

**Comment by KionX** [comment]
Or an engine patch or https://forums.faforever.com/viewtopic.php?f=41&t=13350.html#p169487

**Comment by Garanas** [comment]
What if we spawn dummy entities for units with 'reasonable worthy intel', such as radars, sonars, cruisers and command units, for each ally?

**Comment by Garanas** [comment]
Same for SMDs, use a dummy unit that each player owns but can't select and doesn't render a mesh or anything?

**Comment by KionX** [comment]
I have some developments. I'll try to patch the engine.

**Comment by Garanas** [comment]
👍 , that works too. If that doesn't seem feasible, we can try my approach too - it won't work when you select them, because they'd be unselectable. But it will work when you always render intel rings.

**Comment by HUSSAR-mtrela** [comment]
@Garanas this [forum post](https://forums.faforever.com/viewtopic.php?f=41&t=13350&start=10#p169517) looks like a good way to implement sharing of intel rings and defense rings. Do you agree?

I suggest we limit sharing those rings only to structures (radar, omni, SMD, TMD, SAM) and skip mobile units due to slower performance of computing rings on moving units.

**Comment by Garanas** [comment]
I disagree - there's no need to sync anything with the UI. Alliance changes are managed in the sim. The units are managed in the sim. We can keep this in the sim.

We can create an initial experiment with just the SMD - it is usually the most valuable to know. To me the approach with the dummy units is a great start, include a hook to when alliances are changed and you're good for a prototype.

Performance wise this is all fine - Cybran build drones are dummy units too for example and when approached as a dummy unit they have a little to no cost to the simulation. It would have a similar setup to [[mobile factories]()](https://github.com/FAForever/fa/pull/5227) where we just 'bluntly' attach the dummy units to the SMD. But instead of you owning the dummy unit, you create one for each ally. And those dummy units in turn generate the overlays.
